### ISSUE TITLE
Documentation clarifications

### ISSUE THREAD

**Issue by mole99** [dan-rodrigues/icestation-32#24]
First of all thanks for this awesome project!

So I tried to follow the instructions to run the `build-and-run-sprites-demo.sh` inside `simulator/`.

After compiling the riscv cross-toolchain directly from the [riscv-gnu-toolchain](https://github.com/riscv/riscv-gnu-toolchain) repository I was left with binaries with the prefix `riscv32-unknown-elf-` unlike your `riscv-none-embed-`.

Maybe it can be stated in the documentation that the prefix needs to be changed in `software/common/cross.mk`.

Then after trying to run the script I got the following error:

```
In file included from ../QSPIFlashSim.cpp:7:
../QSPIFlashSim.hpp:108:16: error: ‘string’ in namespace ‘std’ does not name a type
  108 |     const std::string cmd_name(CMD cmd);
      |                ^~~~~~
../QSPIFlashSim.hpp:19:1: note: ‘std::string’ is defined in header ‘<string>’; did you forget to ‘#include <string>’?
   18 | #include <set>
  +++ |+#include <string>

...
```

Which was fixed by adding `#include <string>` in `simulator/QSPIFlashSim.hpp`.

```
#include <stdint.h>
#include <vector>
#include <set>
#include <string>
```

Maybe this line can be added to `QSPIFlashSim.hpp`?

Finally, when I run the simulation I get around 2.4 fps.
This is enough for simple testing but not for playing. Is there a way to speed up the simulation? Verilator supports [multithreaded](https://verilator.org/guide/latest/verilating.html#multithreading) models but when simply passing the flag the screen of the simulation remains black. I guess the other components must be thread safe too?

**Comment by dan-rodrigues** [comment]
Thanks for spotting that error, it’s fixed in a35c004ea0dcb3d7223eb6ff9b0d48f951fb306c. `<string>` was probably being included by one of the other `include` statements on my local setup and that was being accidentally relied on here. The note on the `CROSS` variable was copied form the /software README over to the root level README to make it more prominent.

Simulator speed is a tough one. I mainly used it to quickly verify the hardware and do quick software tests, anything beyond that was done by iterating on an FGPA board. I get a 30% or so speed bump by using the PicoRV32 instead, controlled by this `USE_VEXRISCV ` parameter [here](https://github.com/dan-rodrigues/icestation-32/blob/master/simulator/ics32_tb.v#L51). Simulating the flash was expensive but removing that would limit you to the 64K of RAM, that could be an optional feature to get a few more FPS for small programs.

#14 added a fast way to program a ULX3S without needing to flash which is also convenient for quick iteration, but you'd have to buy one of those first. A compromise would be to make an software emulator that won’t be as accurate as the Verilator sim but you’d get realtime speed. Didn't get around to doing this though.

**Comment by mole99** [comment]
Thanks for the quick fix. I see, I have overlooked the note on the `CROSS` variable in `/software`. I may have been too fast with compiling the example but now that it's moved to the root level it's even clearer.
I already thought that the simulation speed is at it's maximum. It is already amazing that you can simply simulate the entire platform. For testing in hardware I already have an iCEBreaker at home, on which I will try out the examples. Fast flashing isn't that important anyway, just don't mess up and you don't need to reflash ;)
I guess this issue can be closed then, thanks.
